pkg/kubelet/dockershim/network/plugins.go

```diff
@@ -247,18 +248,41 @@ func getOnePodIP(execer utilexec.Interface, nsenterPath, netnsPath, interfaceNam
 	return ip, nil
 }
 
+// preferredFamily will determine the preferred IP family, based on the DNS service IP
+// environment variable (DNS_SVC_IP), if set.
+func preferredFamily() string {
+
+	dnsServiceIP := os.Getenv("DNS_SVC_IP")
+	if dnsServiceIP == "" {
+		glog.V(3).Infof("Unable to determine preferred IP family")
+		return ""
+	}
+	glog.V(3).Infof("DNS Service IP is %s", dnsServiceIP)
+	dnsIP := net.ParseIP(dnsServiceIP)
+	if dnsIP == nil {
+		glog.Warningf("Unable to parse DNS_SVC_IP (%s) to determine preferred family", dnsServiceIP)
+		return "-4"
+	}
+	if dnsIP.To4() == nil {
+		glog.V(3).Infof("Using IPv6 as preferred family")
+		return "-6"
+	}
+	glog.V(3).Infof("Using IPv4 as preferred family")
+	return "-4"
+}
+
 // GetPodIP gets the IP of the pod by inspecting the network info inside the pod's network namespace.
 func GetPodIP(execer utilexec.Interface, nsenterPath, netnsPath, interfaceName string) (net.IP, error) {
-	ip, err := getOnePodIP(execer, nsenterPath, netnsPath, interfaceName, "-4")
-	if err != nil {
-		// Fall back to IPv6 address if no IPv4 address is present
-		ip, err = getOnePodIP(execer, nsenterPath, netnsPath, interfaceName, "-6")
-	}
-	if err != nil {
-		return nil, err
+	addrType := preferredFamily()
+        if addrType != "" {
```

Comment:
small whitespace issue.

Reply:
Will address when I rebase.

Reply:
this whitespace issue seems to be present still.

Reply:
Sorry, I missed it. Had spaces vs tab, so it looked fine in my editor. Will be in next patch.